content/ytt/docs/latest/lang-ref-ytt-schema.md

```diff
@@ -0,0 +1,226 @@
+---
+title: Schema module
+---
+
+## Overview
+
+`ytt` schemas are currently in the **experimental** phase. To use schema features, include `--enable-experiment-schema`.
+
+`ytt`'s Schema feature provides a way to verify data value YAML structures and values with the help of annotations.
+
+A schema allows you to provide default data values, and verify that each data value file provided is type checked and validated against the schema and guaranteed to be proper.
+
+---
+## Defining Schema
+Configuration Authors establish a Schema by capturing the structure in a YAML document with `#@schema/match` at the top:
+```yaml
+#@schema/match data_values=True
+---
+#! Schema contents
+```
+- `data_values` (`bool`) — whether this schema is applicable to data values. If this
+  is not set to `True`, this results in an error.
+
+Notes:
+- Schema must be defined _before_ any other files containing YAML can be processed. Therefore, a file containing a Schema document must not contain other kinds of documents.
+- files containing Schema documents will be detected among input files (i.e. those specified by `-f`).
+- the first file containing a Schema document establishes the "base". Subsequent files containing Schema documents are overlayed onto the "base" in the order they appear (identically to how Data Values files are processed).
+
+---
+## Supported Types
+
+While YAML provides for an extendable range of types, the `ytt` Schema supports a specific set.
+
+### Types
+
+- `bool` — `true`, `false` (and when not strict, `yes`, `no`, `Y`, `N`, etc.)
+- `float` — e.g. `0.4`
+- `int` — e.g. `42`
+- `null` — `null`, `~`, and when value is omitted.
+- `string` — e.g. `""`, `"ConfigMap"`, `"0xbeadcafe"`
+- `any` — any valid YAML value is permitted. Must be set [explicitly](#schematype).
+- `map` — A YAML mapping that must contain exactly one item of each given key, whose value must be of the type specified.
+- `array` — A YAML sequence, each item must be of the specified `type`.
+
+---
+## Inferring Types
+
+Structure in Schema is largely expressed by example rather than by description. Types are — by default — inferred based on the values given. The Configuration Author can override these defaults using the [`@schema/type`](#schematype) annotation.
```

Comment:
```suggestion
Structure in Schema is largely expressed by example rather than by description. Types are — by default — inferred based on the values given.
```
As of today (and slated for Schemas v1), setting the type explicitly is **not** supported. Only `@schema/type any=True` is supported.